README.md

```diff
@@ -30,22 +28,50 @@ Use the `-port` option to specify the port number for the server to listen reque
 ./code-runner-server -port <port>
 ```
 
-## Send code files and arguments to server
-Use curl command to connect to the server and send request.
-Use multipart/form data option(-F) to provide the assignment submission file and the command-line arguments that are to be used when executing the code.
+## Code Runner as a docker image
+- A docker image of code runner is used as a base image for different language specific images.
+- Code Runner is not directly run, instead use the desired language specific image to run the assignments.
+- Prerequisite for this is that docker engine should be installed.
+- See [instructions](https://docs.docker.com/engine/installation/) for installing docker engine on different supported platforms.
+
+Use the following docker command to run the language specific image.
 ```commandline
-curl <server_ip_address:port_number> -F <compressed_filepath> -F <arg1> -F <arg2> ...
+docker run --publish <port_to_expose>:<port_to_run> <language_image_tag> -port <port_to_run>
 ```
 
+## Supported Languages
+A fixed set of languages and their versions are currently supported.
+- gcc 7
+- g++ 7
+- python 3.7
+- java 8 & 11
+
 ## Web page to Submit Assignment
-![](docs/codeRunnerPage.png)
+Use the exposed port number to open the web page. Example - for exposed port number as 52453, hit `localhost:52453` on browser.
+
+![](docs/webPage.png)
 
 Submit your assignment through the web page as shown in the figure above.
+- Initially `Build` and `Run` buttons are disabled. Submit the assignment details to enable them.
 - Choose the assignment tar ball.
 - Enter commands to compile and run the code.
-- Enter working directory to run the commands. If not provided then commands will be run at the root assignment directory.
+- Enter working directory to run the commands. If not provided then commands will be run at the assignment root directory.
 - Add command line arguments if required to run the assignment using `Add` button.
 - Click on `Submit` button to upload the details to the server.
 - The status of upload will be displayed in the `Output` section.
-- Click on `Build & Run` button to run the code.
+- Click on `Build` button to compile the code.
+- The status of build will be displayed in the `Output` section.
+- Click on `Run` button to run the compiled code.
 - The output will be displayed in the `Output` section as shown above.
+
+#### Add Command Line Arguments
+- Command Line Arguments can be added as key-value pairs in the text boxes.
+- First text box is for key and second is for the value. 
+- If the arguments have only value then use second text box for the value. First can be left empty.
```

Comment:
If specifying only the value, then use the text box on the right hand side.

Reply:
Done